processing/src/main/java/org/apache/druid/query/filter/RangeFilter.java

```diff
@@ -812,6 +826,25 @@ public String toString()
     }
   }
 
+  /**
+   * if casting to a LONG typed ARRAY, first cast to a DOUBLE typed ARRAY and then {@link Math#ceil(double)} on the
+   * values to ensure a correct lower bound
+   */
+  private static Object[] castArrayToCeil(ExprEval<?> lowerEval, ExpressionType inputType)
+  {
+    if (ExpressionType.LONG_ARRAY.equals(inputType)) {
+      final ExprEval<?> doubleArray = lowerEval.castTo(ExpressionType.DOUBLE_ARRAY);
```

Comment:
This is going to lose some fidelity if the input values are numeric strings that cannot be represented as doubles. To maintain fidelity, it would be better to cast each value individually, trying long first (using `GuavaUtils.tryParseLong`) and then using double only if the coercion to long can't be done.

Reply:
i believe that is effectively what will happen, expression casting to numbers uses https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/math/expr/ExprEval.java#L625 which does that